mountpoint-s3/src/data_cache/disk_data_cache.rs

```diff
@@ -230,11 +230,12 @@ impl DiskDataCache {
         block_idx: BlockIndex,
         block_offset: u64,
     ) -> DataCacheResult<Option<ChecksummedBytes>> {
-        let mut file = match fs::File::open(path.as_ref()) {
+        let file = match fs::File::open(path.as_ref()) {
             Ok(file) => file,
             Err(err) if err.kind() == ErrorKind::NotFound => return Ok(None),
             Err(err) => return Err(err.into()),
         };
+        let mut file = BufReader::new(file);
```

Comment:
Would increasing the default buffer size of 8 KiB help here? As we know our blocks are 1 MiB, higher buffer size might be better?

Reply:
That's a really good question, I hadn't actually considered how it decides how much to buffer and how the size of that buffer can help. I'll need to think about it a bit more and give some different values a try.

Reply:
Thinking about this a bit more critically, maybe we shouldn't use a buffered reader at all. What we really do here is read the whole chunk in no matter what, so we should just load the whole thing into memory allocating it up front (1MiB + a little bit for the header).

Reply:
From a very quick look at the `BufReader` code: https://doc.rust-lang.org/src/std/io/buffered/bufreader.rs.html#281, the small (default) buffer may actually be good enough to read the header while the 1MiB block skips it.

But agree to measure performance before changing anything.